Add floor selector (#32124)
home-assistant/home-assistant.github.io: source/_docs/blueprint/selectors.markdown

      device_class: motion
```

## Floor selector

> The floor selector shows a floor finder that can pick a single or multiple
> floors based on the selector configuration. The value of the input will be the
> floor ID, or a list of floor IDs, based on if `multiple` is set to `true`.

Reviewer: ```suggestion
The floor selector shows a floor finder that can pick 
floors based on the selector configuration. The value of the input will be the
floor ID. If `multiple` is set to `true`, the value is a list of floor IDs.
```

> A floor selector can filter the list of floors, based on properties of the
> devices and entities that are assigned to the areas on those floors.
> For example, the floor list could be limited to floors with entities
> provided by the [ZHA](/integrations/zha) integration that are on the floor
> based on the area they are in.

Reviewer: ```suggestion
A floor selector can filter the list of floors based on the properties of the
devices and entities assigned to the areas on those floors.
For example, the floor list could be limited to floors with entities
provided by the [ZHA](/integrations/zha) integration, based on the areas they are in.
```

Reviewer: Tried a rephrase to simplify the sentence a bit. If I misunderstood the meaning, please ignore my comment.

Author: It will not "limit further", the area is just the container it searches (as devices & entities are assigned to areas, not floors).

Reviewer: oh, I see. I rephrased the rephrase :-D.

> In its most basic form, this selector doesn't require any options,
> which will show all floors.

Reviewer: ```suggestion
In its most basic form, this selector doesn't require any options.
It will show all floors.
```

![Screenshot of a floor selector](/images/blueprints/selector-floor.png)

```yaml
floor:
```

{% configuration floor %}
device:
  description: >

>     When device options are provided, the list of floors is filtered by floor
>     that at least have one device that matches the given conditions. Can be
>     either a object or a list of object.

Reviewer: ```suggestion
    When device options are provided, the list of floors is filtered by floors
    that have at least one device matching the given conditions. Can be
    either an object or a list of objects.
```

  type: list
  required: false
  keys:
    integration:
      description: >

>         Can be set to an integration domain. Limits the list of floor that
>         have devices by the set integration domain, for example,
>         [`zha`](/integrations/zha).

Reviewer: ```suggestion
        Can be set to an integration domain. Limits the list of floors that
        have devices by this integration domain. For example,
        [`zha`](/integrations/zha).
```

      type: string
      required: false
    manufacturer:
      description: >

>         When set, it limits the list of floor that have devices by the set
>         manufacturer name.

Reviewer: ```suggestion
        When set, the list only includes floors that have devices by the set
        manufacturer name.
```

      type: string
      required: false
    model:
      description: >

>         When set, it limits the list of floors that have devices which have
>         the set model.

Reviewer: ```suggestion
        When set, the list only includes floors that have devices which have
        the set model.
```

      type: string
      required: false
entity:
  description: >

>     When entity options are provided, the list of floors is filtered by floors
>     that at least have one entity that matches the given conditions. Can be
>     either a object or a list of object.

Reviewer: ```suggestion
    When entity options are provided, the list only includes floors
    that at least have one entity that matches the given conditions. Can be
    either an object or a list of objects.
```

  type: list
  required: false
  keys:
    integration:
      description: >

>         Can be set to an integration domain. Limits the list of floors that
>         have entities by the set integration domain, for example,

Reviewer: ```suggestion
        Can be set to an integration domain. Limits the list of floors that
        have entities by the set integration domain. For example,
```

        [`zha`](/integrations/zha).
      type: string
      required: false
    domain:
      description: >

>         Limits the list of floors that have entities of a certain domain(s),
>         for example, [`light`](/integrations/light) or
>         [`binary_sensor`](/integrations/binary_sensor). Can be either a string
>         with a single domain, or a list of string domains to limit the selection to.

Reviewer: ```suggestion
        When set, the list only includes floors that have entities of certain domains,
        for example, [`light`](/integrations/light) or
        [`binary_sensor`](/integrations/binary_sensor). Can be either a string
        with a single domain, or a list of string domains to limit the selection to.
```

      type: [string, list]
      required: false
    device_class:
      description: >

>         Limits the list of floors to floors that have entities with a certain
>         device class(es), for example, `motion` or `window`. Can be either a string
>         with a single device_class, or a list of string device_class to limit
>         the selection to.

Reviewer: ```suggestion
        When set, the list only includes floors that have entities with a certain
        device class, for example, `motion` or `window`. Can be either a string
        with a single device_class, or a list of string device_class to limit
        the selection.
```

      type: [device_class, list]
      required: false
    supported_features:
      description: >

>         Limits the list of floors to floors that have entities with a certain
>         supported feature, for example, `light.LightEntityFeature.TRANSITION`
>         or `climate.ClimateEntityFeature.TARGET_TEMPERATURE`. Should be a list
>         of features.

Reviewer: ```suggestion
        When set, the list only includes floors that have entities with a certain
        supported feature, for example, `light.LightEntityFeature.TRANSITION`
        or `climate.ClimateEntityFeature.TARGET_TEMPERATURE`. Should be a list
        of features.
```

      type: list
      required: false
multiple:
  description: >
    Allows selecting multiple floors. If set to `true`, the resulting value of
    this selector will be a list instead of a single string value.
  type: boolean
  default: false
  required: false
{% endconfiguration %}

The output of this selector is the floor ID, or (in case `multiple` is set to
`true`) a list of floor IDs.

```yaml
# Example floor selector output result, when multiple is set to false
first_floor

# Example floor selector output result, when multiple is set to true
- first_floor
- second_floor
```

### Example floor selectors <!-- omit from toc -->

An example floor selector only shows floors that have one or more lights or
switches provided by the [ZHA](/integrations/zha) integration.

```yaml
floor:
  entity:
    integration: zha
    domain:
      - light
      - switch
```

> Another example uses the are floor selector, which only shows floors that

Reviewer: ```suggestion
Another example uses the are floor selector, which only shows floors that
```
Something went wrong here. Not sure if it's just a typo or if the idea was to refer to a specific floor selector?

Author: ```suggestion
Another example using the floor selector, which only shows floors that
```
